.github/workflows/ci-run-test.yml

```diff
@@ -69,7 +69,6 @@ env:
   PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_PROD_ACCESS_TOKEN }}
   # Release builds use the service, PR checks and snapshots will use the local backend if possible.
   PULUMI_TEST_USE_SERVICE: ${{ !contains(inputs.version, '-') }}
-  PULUMI_TEST_PYTHON_SHARED_VENV: "true"
```

Comment:
The reason for this is to speed up CI, so the tests aren't always recreating venvs over and over.

Do we know if there are a specific set of tests that are affected by this? If so, wondering if we should unset this envvar just for those tests.